File: C# Advanced/DefiningClasses-Exercise/06.SpeedRacing/Car.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace DefiningClasses

{
    public class Car
    {
        public Car(string model, double fuelAmount, double fuelConsumption)
        {
            this.Model = model;
            this.FuelAmount = fuelAmount;
            this.FuelConsumptionPerKilometer = fuelConsumption;

        }
        public string Model { get; set; }

        public double FuelAmount { get; set; }

        public double FuelConsumptionPerKilometer { get; set; }

        public double TravelledDistance { get; set; }

        public bool Drive(double distanceTraveled)
        {
            double neededFuel = distanceTraveled * FuelConsumptionPerKilometer;

            if (neededFuel > FuelAmount)
            {
                return false;
            }

            this.FuelAmount -= neededFuel;
            this.TravelledDistance += distanceTraveled;
            return true;
        }
    }
}

File: Java Advanced/DefiningClasses-Exercises/src/SpeedRacing_03/Car.java
package SpeedRacing_03;

public class Car {
    private String model;
    private double fuelAmount;
    private double fuelNeededPerKm;
    private int distanceTraveled;

    public Car(String model, double fuelAmount, double fuelNeededPerKm) {
        this.model = model;
        this.fuelAmount = fuelAmount;
        this.fuelNeededPerKm = fuelNeededPerKm;
        this.distanceTraveled = 0;
    }

    public String getModel() {
        return model;
    }

    public double getFuelAmount() {
        return fuelAmount;
    }

    public int getDistanceTraveled() {
        return distanceTraveled;
    }

    public double getFuelNeededPerKm() {
        return fuelNeededPerKm;
    }

    public boolean drive(int kmToDrive) {
        double neededFuel = kmToDrive * getFuelNeededPerKm();
        if (neededFuel > getFuelAmount()){
            return false;
        }
        this.distanceTraveled += kmToDrive;
        this.fuelAmount -= neededFuel;

        return true;
    }

    @Override
    public String toString() {
        return String.format("%s %.2f %d", getModel(), getFuelAmount(), getDistanceTraveled());
    }
}
